dist/run.sh

```diff
@@ -44,6 +44,24 @@ copy_jars() {
     fi
 }
 
+fix_permissions() {
+    path="$1"
+    if [ -e "$path" ]; then
+        if [ -d "$path" ]; then
+            chmod 0755 $path
+            for child in $path/*; do
+                fix_permissions "$child"
+            done
+        else
+            if [ -x "$path" ]; then
```

Comment:
`-x` might have its own complications. From `man test`:
> -x FILE
>               FILE exists and execute (or search) permission is granted
Seems to me the result will depend on the user running the test.

Reply:
Yes, this checks if the current user can execut the file. Would be probably cleaner to check the actual execution bit, but I have no idea how, except of of parsing the output of `ls -l $child`, which is rather dirty. Do you have better ideas?

Reply:
`stat -c '%a' "$path"` and check if last three numbers are odd? Probably too much work.

Reply:
Changed to 
```shell
all_permissions=$(stat --format "%a" $path)
user_permissions=${all_permissions: -3 : 1}
user_exec_bit=$(( $user_permissions % 2 ))
if [ $user_exec_bit -ne 0 ]; then
...
```

Reply:
The above solution, unfortunately is not compatible with OSX version of `stats`. I revert to the use of `if [ -x" $path" ]; then`.